src/reanimated/ReanimatedNativeStackScreen.tsx

```diff
@@ -30,24 +32,29 @@ const ReanimatedNativeStackScreen = React.forwardRef<
   typeof AnimatedScreen,
   ScreenProps
 >((props, ref) => {
+  const { options } = useScreenInfo();
+
   const { children, ...rest } = props;
   const { stackPresentation = 'push' } = rest;
 
   const dimensions = useSafeAreaFrame();
   const topInset = useSafeAreaInsets().top;
-  let statusBarHeight = topInset;
-  const hasDynamicIsland = Platform.OS === 'ios' && topInset === 59;
-  if (hasDynamicIsland) {
-    // On models with Dynamic Island the status bar height is smaller than the safe area top inset.
-    statusBarHeight = 54;
-  }
+  const isStatusBarTranslucent = rest.statusBarTranslucent ?? false;
+  const statusBarHeight = getStatusBarHeight(
+    topInset,
+    dimensions,
+    isStatusBarTranslucent
+  );
+
+  const isLargeHeader = options.headerLargeTitle ?? false;
```

Comment:
Why do we apply this fix only for `ReaniamtedNativeStackScreen`? Where does the difference come from?

Reply:
This is applied just for the compliance between the NativeStackView and ReanimatedNativeStackScreen - I forgot to apply it here earlier 😅 Take a look on the [NativeStackView#L225](https://github.com/software-mansion/react-native-screens/blob/24689052c009f383657a74521c5ce875044ee2ef/src/native-stack/views/NativeStackView.tsx#L225).